frontend/src/lib/MlmdUtils.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  Context,
+  Api,
+  Execution,
+  getResourceProperty,
+  ExecutionProperties,
+  ExecutionCustomProperties,
+} from '@kubeflow/frontend';
+import {
+  GetContextByTypeAndNameRequest,
+  GetExecutionsByContextRequest,
+} from '@kubeflow/frontend/src/mlmd/generated/ml_metadata/proto/metadata_store_service_pb';
+
+async function getContext({ type, name }: { type: string; name: string }): Promise<Context> {
+  const request = new GetContextByTypeAndNameRequest();
+  request.setTypeName(type);
+  request.setContextName(name);
+  try {
+    const res = await Api.getInstance().metadataStoreService.getContextByTypeAndName(request);
+    const context = res.getContext();
+    if (context == null) {
+      throw new Error('Cannot find specified context');
+    }
+    return context;
+  } catch (err) {
+    err.message = `Cannot find context with ${JSON.stringify(request.toObject())}: ` + err.message;
+    throw err;
+  }
+}
+
+/**
+ * @throws error when network error, or not found
+ */
+export async function getTfxRunContext(argoWorkflowName: string): Promise<Context> {
```

Comment:
Actually this remind me of one thing. See https://github.com/tensorflow/tfx/blob/5ee3de9c77a92e904010c7656d6269d3b28744bf/tfx/orchestration/kubeflow/base_component.py#L136

Users can actually override the TFX run id to be something different than Argo workflow ID, which might cause problem to this piece of logic. 

@neuromage Not sure how useful that customization is currently. Shall we consider closing the door of overriding run_id?

Reply:
Thanks! that's a good catch.
From the comment
```
# Allow overriding pipeline's run_id externally, primarily for testing.
```
looks like it was for testing.

Reply:
> Actually this remind me of one thing. See https://github.com/tensorflow/tfx/blob/5ee3de9c77a92e904010c7656d6269d3b28744bf/tfx/orchestration/kubeflow/base_component.py#L136
> 
> Users can actually override the TFX run id to be something different than Argo workflow ID, which might cause problem to this piece of logic.
> 
> @neuromage Not sure how useful that customization is currently. Shall we consider closing the door of overriding run_id?

Can we add something explicit like `workflow_name` that's not overridable?

Reply:
> Can we add something explicit like workflow_name that's not overridable?

Do you mean here or on the TFX side?

Reply:
On the TFX side.

Reply:
> On the TFX side.

Yes. I think I'll make a CL closing the door.